src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -247,32 +247,34 @@ public boolean checkSolidity(Hash hash, boolean milestone, int maxProcessedTrans
         if(fromHash(tangle, hash).isSolid()) {
             return true;
         }
-        Set<Hash> analyzedHashes = new HashSet<>(snapshotProvider.getInitialSnapshot().getSolidEntryPoints().keySet());
+        Set<Hash> analyzedHashes = new LinkedHashSet<>(snapshotProvider.getInitialSnapshot().getSolidEntryPoints().keySet());
         if(maxProcessedTransactions != Integer.MAX_VALUE) {
             maxProcessedTransactions += analyzedHashes.size();
         }
         boolean solid = true;
         final Queue<Hash> nonAnalyzedTransactions = new LinkedList<>(Collections.singleton(hash));
         Hash hashPointer;
         while ((hashPointer = nonAnalyzedTransactions.poll()) != null) {
-            if (analyzedHashes.add(hashPointer)) {
-                if(analyzedHashes.size() >= maxProcessedTransactions) {
-                    return false;
-                }
+            if (!analyzedHashes.add(hashPointer)) {
+                continue;
+            }
+
+            if (analyzedHashes.size() >= maxProcessedTransactions) {
+                return false;
+            }
+
+            final TransactionViewModel transaction = fromHash(tangle, hashPointer);
```

Comment:
Btw, since you started refactoring you can feel free to remove the `final`

No need for it when you have a small local scope

Reply:
Removed.